week02/ValidParentheses.cpp

```diff
@@ -0,0 +1,24 @@
+class Solution {
+public:
+    bool isValid(string s) {
+        //TC : O(N)
+        //MC : Confused(?)
+        stack<char>bracketStack;
+        for(auto character : s){
+            if(character == '(' || character == '{' || character == '['){
+                bracketStack.push(character);
+            }
+            else if(character == ')' || character == '}' || character == ']'){
+                if(!bracketStack.empty()){
+                    if((character == ')' && bracketStack.top() == '(') || (character == '}' && bracketStack.top() == '{') || (character == ']' && bracketStack.top() == '[')){
+                        bracketStack.pop();
+                    }
```

Comment:
Too large chaining, so many `if-else`. I would suggest you check the official solution after I post it.